File: Python/Queue.py
class Node:
    def __init__(self, data, node=None):
        self.data = data
        self.link = node
    
    def set_link(self, node):
        self.link = node

    def get_link(self):
        return self.link

class Queue:
    def __init__(self):
        self.front = None
        self.back = None

    def enqueue(self, data):
        temp = Node(data)
        if self.front == None:
            self.back = temp
            self.front= self.back
        else:
            temp.link = self.back
            self.back = temp

    def dequeue(self, data):
        if self.front == None:
            print("Queue Empty!")
        val = self.front.data
        self.front = self.front.link
        return val

    def peek(self):
        if self.front == None:
            print("Queue Empty!")
        return self.front.data 

    def isEmpty(self):
        if self.front == None:
            return True
        return False

    def print_list(self):
        temp = self.front
        while temp != None:
            print(temp.data)
            temp = temp.link

    def size(self):
        temp = self.front
        counter = 0
        while temp != None:
            temp = temp.link
            counter += 1
        return counter

def main():
    print("Hello, There!")
    
    queue = Queue()

    print(queue.isEmpty())
    queue.enqueue(1)
    queue.enqueue(3)
    queue.enqueue(5)
    queue.enqueue(7)
    queue.enqueue(9)
    
    print(queue.peek())

    queue.print_list()
    print(queue.denqueue())
    print(queue.isEmpty())
    queue.print_list()


if __name__ == "__main__":
    main()

File: Javascript/Queue.js
class Node {
    constructor(data, node){
        this.data = data;
        this.link = node;
    }

    setLink = (node) => {
        this.link = node;
    }

    getLink = () => {
        return this.link;
    }
}

class Queue {
    constructor(){
        this.front = null;
        this.back = null;
    }

    enqueue = (data) => {
        let temp = new Node(data, null)
        if (this.front === null){
            this.back = temp;
            this.front = this.back;
        } else {
            temp.link = this.back;
            this.back = temp;
        }
    }

    peek = () => {
        let temp = this.front;
        if (temp === null){
            console.log("Queue Empty!");
        }
        return temp.data;
    }

    dequeue = () => {
        if (this.front === null){
            console.log("Queue Empty!");
        }
        let data = this.front.data;
        this.front = this.front.link;
        return data;
    }

    isEmpty = () => {
        if (this.front === null){
            return true;
        } else {
            return false;
        }
    }

    size = () => {
        let temp = this.front;
        let counter = 0;
        while (temp != null){
            counter += 1;
            temp = temp.link;
        }
        return counter;
    }

    printList = () => {
        let temp = this.front;
        while (temp != null){
            console.log(temp.data);
            temp = temp.link;
        }
    }
}

let queue = new Queue();

console.log(queue.isEmpty());

queue.enqueue(1);
queue.enqueue(3);
queue.enqueue(5);
queue.enqueue(7);
queue.enqueue(9);

console.log(queue.isEmpty());

queue.printList();
console.log(queue.size());

console.log(queue.peek());
console.log(queue.dequeue());
console.log(queue.peek());
queue.printList();
